tests/neg/i2473.scala

```diff
@@ -0,0 +1,7 @@
+trait Foo { trait Inner }
+trait Bar { foo: Foo =>
+  type Inner <: foo.Inner // error: type Inner cannot have the same name as trait Inner in trait Foo -- class definitions cannot be overridden
```

Comment:
Oh, yes. I reused the old error message. I will improve it to make the distinction.